controllers/flowlogspipeline/flp_objects.go

```diff
@@ -510,6 +540,48 @@ func (b *builder) service(old *corev1.Service) *corev1.Service {
 	return newService
 }
 
+func (b *builder) promService(old *corev1.Service) *corev1.Service {
+	if old == nil {
```

Comment:
I see that later, `promService` method is invoked with a non-nil object and also explicitly `nil`.

Could be this method explicitly divided into two? e.g.:
```
func (b *builder) newPromService() *corev1.Service
func (b *builder) updatedPromService(old *corev1.Service) *corev1.Service
```